lib/SILOptimizer/PassManager/PassManager.cpp

```diff
@@ -729,6 +729,56 @@ void SILPassManager::addFunctionToWorklist(SILFunction *F,
 
   StoredLevel = NewLevel;
   FunctionWorklist.push_back(F);
+  // Update FunctionWorklist such that all the callees in function 'F' are
+  // present after it.
+  updateFunctionWorklist(F);
+}
+
+void SILPassManager::updateFunctionWorklist(SILFunction *F) {
```

Comment:
This won't work as is with BottomUpFunctionOrder because we won't be examining the callers of the newly added functions.

Reply:
Sure, but I think with a small modification of BottomUpFunctionOrder, if could be used for this purpose as well: Instead of iterating over all functions in FindSCCs, just call DFS for the newly added function.